src/main/java/uk/gov/justice/digital/delius/service/ReferenceDataService.java

```diff
@@ -62,4 +68,19 @@ private StandardReference pomAllocationTransferReason(String reason) {
         return standardReferenceRepository.findByCodeAndCodeSetName(reason, POM_ALLOCATION_REASON_DATASET)
                 .orElseThrow(() -> new RuntimeException(String.format("No pom allocation reason found for %s", reason)));
     }
+
+    public Page<KeyValue> getProbationAreasCodes(boolean restrictActive) {
+        final var filter = ProbationAreaFilter.builder().restrictActive(restrictActive).build();
+
+        return probationAreaRepository.findAll(filter).stream()
+                .map(area -> new KeyValue(area.getCode(), area.getDescription()))
+                .collect(collectingAndThen(toList(), PageImpl::new));
+    }
+
+    public Page<KeyValue> getLocalDeliveryUnitsForProbationArea(String code) {
+        return probationAreaRepository.findByCode(code).stream()
```

Comment:
Slightly confused why this returns a List I would have thought it would be an Optional
`probationAreaRepository.findByCode(code)`

Reply:
[stream](https://docs.oracle.com/javase/9/docs/api/java/util/Optional.html#stream--) is a new method on Optional